### PR TITLE
lua: fix compilation on Cray

### PR DESCRIPTION
**Pull request by SteVwonder**
Fixes lua+shared compilation on a cray system.  See commit message for details.

### CODE REVIEW

File: var/spack/repos/builtin/packages/flux-core/package.py [hunk 1]
```diff
@@ -46,6 +48,7 @@ class FluxCore(AutotoolsPackage):
     depends_on("python", type=('build', 'run'))
     depends_on("python@2.7:2.99", when="@0.1.0:0.11.0")
     depends_on("python@2.7:", when="@0.11.1:")
+    depends_on("python@3.6:", when="@0.17.0:")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/flux-core/package.py]: You should add `type=('build', 'run')` to all of these python deps.

**Comment by adamjstewart** [comment]
It would probably be quicker to review and merge if these changes were all in separate PRs. For example, the new versions I can merge right away, but the Linux/Cray change will take longer to test.

**Comment by SteVwonder** [comment]
Ok.  Just dropped the flux version bump from this PR.  I'll post a new PR for that soon, with your suggested changes included. Thanks!

**Review by alalazo — APPROVED** [review]

**Review by adamjstewart — APPROVED** [review]

**Comment by SteVwonder** [comment]
Thanks!

**Pull request merged**